### PR TITLE
Allow greater flexility in configuring redis cache and client

### PR DESCRIPTION
**Pull request by jrcamp**
This allows a configuration value other than `redis_host` to be used. Also
allow a `RedisClient` instance to be passed to a RedisCache instead of also
being hard coded.

(I'd argue that `RedisCache` should never instantiate `RedisClient` and it
should always be passed through as a parameter manually for via DI.)

### CODE REVIEW

File: cache-utils/src/main/java/io/ndustrial/cacheutils/RedisClient.java [hunk 1]
```diff
@@ -16,16 +16,26 @@
...
     protected int _database;
 
-    public RedisClient(Map conf, int db)
+    /**
+     * Constructor
+     * @param conf configuration map
+     * @param hostKey hostKey inside configuration map to pull host from
+     * @param db db number
+     */
+    public RedisClient(Map conf, String hostKey, int db)
```
Comment (borregoho) [inline cache-utils/src/main/java/io/ndustrial/cacheutils/RedisClient.java]: can we have RedisClient(String host, int db), just to skip the config map?

**Comment by borregoho** [comment]
I like the DI approach, is it to much of a refactor to go that way?

File: cache-utils/src/main/java/io/ndustrial/cacheutils/RedisClient.java (hunk 1 above)
Comment (jrcamp) [reply cache-utils/src/main/java/io/ndustrial/cacheutils/RedisClient.java]: Yeah I think that's probably the cleanest way. And port as well.

**Comment by jrcamp** [comment]
Looks like we're all in agreement on passing things explicitly/DI. I'd love to refactor it but I have higher priority stuff at the moment.

**Pull request merged**